scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowDoFns.scala

```diff
@@ -60,22 +57,20 @@ sealed trait PredictDoFn[T, V, M <: Model[_]]
 
   def outputTensorNames: Seq[String]
 
-  override def createResource(): Resource[M] = new ConcurrentHashMap[String, (AtomicInteger, M)]()
+  override def createResource(): Resource[M] = new ConcurrentHashMap[String, (Int, M)]()
 
   override def getResourceType: DoFnWithResource.ResourceType = ResourceType.PER_CLASS
 
   @Setup
   override def setup(): Unit = {
     super.setup()
-    val (a, _) = getResource.computeIfAbsent(
+    getResource.compute(
       modelId,
-      new Function[String, (AtomicInteger, M)] {
-        override def apply(v1: String): (AtomicInteger, M) =
-          new AtomicInteger(0) -> loadModel
+      {
```

Comment:
This doesn't hurt to move the two ops in one.

I think this fundamentally looks same as `computeIfAbsent` and `a.incrementAndGet` called in sequence.
In the previous implementation as well only one thread out of many could have created `computeIfAbsent` and rest of the parallel threads would have incremented the count on line 77.

Do we have any example of a failed test in parallel execution? Has any bug been filed for this.

Reply:
> Do we have any example

See example in this comment: https://github.com/spotify/scio/pull/5011#pullrequestreview-1639305224

The `setup` thead can access an element which would be then be deleted by `teardown` if there is a race in between the `computeIfAbsent` and  `incrementAndGet`.
this is very unlikely and we've never had any failure nor report.

Reply:
Ok we moved the `loadModel()` and `close()` to a critical section. Makes sense.